### DISCUSSION TITLE
TypeORM and MySQL

### DISCUSSION THREAD

**Discussion by Ahmadh26** [nextauthjs/next-auth#4322] [Help]
Hey, I'm using next-auth with nextjs, anyone able to help me out with using adapters in order to connect next-auth with my MySQL database? 

I have been looking at a lot of guides and right now i'm stuck at this: 
I login using github or google and get this error in the console: 
```bash
[next-auth][error][adapter_error_getUserByAccount] 
https://next-auth.js.org/errors#adapter_error_getuserbyaccount You must provide selection conditions in order to find a single row. {
  message: 'You must provide selection conditions in order to find a single row.',
  stack: 'Error: You must provide selection conditions in order to find a single row.\n' +
    '    at EntityManager.findOne (C:\\Users\\a_4n\\Desktop\\repos\\ns-v2\\node_modules\\typeorm\\entity-manager\\EntityManager.js:539:19)\n' +
    '    at getUserByAccount (C:\\Users\\a_4n\\Desktop\\repos\\ns-v2\\node_modules\\@next-auth\\typeorm-legacy-adapter\\dist\\index.js:96:37)\n' +
    '    at runMicrotasks (<anonymous>)\n' +
    '    at processTicksAndRejections (node:internal/process/task_queues:96:5)',
  name: 'Error'
}
```

here is my adapter config: 

```ts
adapter: TypeORMLegacyAdapter({
    type: 'mysql', // or mysql, postgresql, mssql
    database: process.env.mysql_database,
    synchronize: true,
    host:  process.env.mysql_host,
    port:  process.env.mysql_port,
    username:  process.env.mysql_username,
    password:  process.env.mysql_password,
    entities: ['../../../lib/entities/AccountEntity.ts'],
    namingStrategy: new SnakeNamingStrategy(),
  }),
```

i do not know what i'm doing wrong tbh .. i want to use a database to save sessions and retrieve them, also would want to add a credentials provider.

**Comment by dixakman** [comment]
+++

**Comment by CGsama** [comment]
Got here by search the error message, it happen to me after some update on TyoeORM. Try put search criteria of findOne in a where object, like findOne({where: {xx:yy}}) instead of findOne({xx:yy}) or use findOneBy({xx:yy})

**Reply by kaliprasadkandula** [reply]
Thanks it worked

**Comment by arnaudjnn** [comment]
Hi guys,

I have exactly the same issue by trying the v4 migration with `TypeORMLegacyAdapter`

**Reply by kwiat1990** [reply]
Is there any solution to fix this error or for now only a downgrade of `typeorm` package? (Updted from `0.2.45` to `0.3.6`)

**Reply by Mayur-505** [reply]
@kwiat1990  answer by @CGsama  worked for me

**Comment by gentunian** [comment]
hi, for everyone to have context, @arnaudjnn is right, and the change was introduced here:

https://github.com/typeorm/typeorm/pull/8801

I created a comment just to make a point, but this sadly was already merged.
